File: doc/src/codes/Users_Manual/geometry_2.py
import numpy as np
from pyautd3 import AUTD3, Controller
from pyautd3.geometry import EulerAngles, rad

Controller.builder()\
    .add_device(AUTD3([0.0, 0.0, 0.0]))\
    .add_device(
        AUTD3([AUTD3.device_width(), 0.0, 0.0]).with_rotation(
            EulerAngles.from_zyz(0 * rad, np.pi / 2 * rad, 0 * rad)))

File: doc/src/codes/Users_Manual/geometry_2.rs
# extern crate autd3;
# extern crate tokio;
# use autd3::prelude::*;
# 
# #[tokio::main]
# async fn main() -> Result<(), Box<dyn std::error::Error>> {
let autd = Controller::builder()
    .add_device(AUTD3::new(Vector3::zeros()))
    .add_device(
        AUTD3::new(Vector3::new(0., 0., AUTD3::DEVICE_WIDTH))
            .with_rotation(EulerAngle::ZYZ(0. * Rad, PI/2.0 * Rad, 0. * Rad)))
#    .open_with(autd3::link::Nop::builder()).await?;
# Ok(())
# }